helm-chart/binderhub/templates/image-cleaner.yaml

```diff
@@ -33,6 +33,13 @@ spec:
         key: hub.jupyter.org_dedicated
         operator: Equal
         value: user
+      {{- range .Values.tolerations }}
```

Comment:
Using `{{- with .Values.tolerations }} ... {{.}} ...` is a nicer way to do this instead of repeating all keys:
https://github.com/jupyterhub/binderhub/blob/ed61e2fb73142a0c3ccf01877c3ad1de83ce6d59/helm-chart/binderhub/templates/deployment.yaml#L44-L47

Reply:
That does not append to the existing tolerations already defined in image-cleaner.yaml

Reply:
You can do it still but without wrapping the `tolerations:` part in the with clause in such situations.

I appreciate that this method decouples us from needing to know the k8s spec of tolerations. One could for example omit one of the fields i think, but they may as the template is become  now be required.

Reply:
You want to conditionally render the tolerations as yaml if they are non-empty, so include a if or with clause for this as well.

The with makes you able to not repeat .Values.tolerations and instead write just 
 and is what we often use

Reply:
Does that work for you? Because it doesn't for me.

With:

```
      tolerations:
      - effect: NoSchedule
        key: hub.jupyter.org/dedicated
        operator: Equal
        value: user
      - effect: NoSchedule
        key: hub.jupyter.org_dedicated
        operator: Equal
        value: user
      {{- with .Values.tolerations }}
      {{- . | toYaml | nindent 8 }}
      {{- end }}
```

```sh
$ helm template . -f values.yaml
Error: YAML parse error on binderhub/templates/image-cleaner.yaml: error converting YAML to JSON: yaml: line 28: did not find expected key
```

Reply:
You might need a different indentation level, it'll need to match the indentation of the existing tolerations. E.g. try `6`?

Reply:
Spot on 👍🏻